### ISSUE TITLE
Clicking an a tag with an href that is supposed to go nowhere, navigates anyway

### ISSUE THREAD

**Issue by retrodaredevil** [grafana/grafana#85044]
### What happened?

I first reported an issue in GraphiQL here: https://github.com/graphql/graphiql/issues/3565 - I am trying to embed a GraphiQL editor for a new datasource I am creating.

There are a few places in GraphiQL that dynamically create a tag similar to this: `<a href="javascript:void 0">lol</a>`. The intent behind this code is that the element should behave like a link, but should not actually go anywhere

Clicking on `<a href="javascript:void 0">lol</a>` navigates me to http://localhost:3000/javascript:void%200

### What did you expect to happen?

Clicking on `<a href="javascript:void 0">lol</a>` should not navigate me away from the page.

### Did this work before?

Maybe? This functionality was probably introduced to Grafana at some point, as if you were to put `<a href="javascript:void 0">lol</a>` on a vanilla HTML page, you are not navigated away.

### How do we reproduce it?

1. In a datasource query editor of your choice, add `<a href="javascript:void 0">lol</a>` as an element.
2. Click "lol" or the text inside the a tag
3. You are navigated away from the page

### Environment (with versions)?

Grafana: Grafana v10.3.3 (252761264e)
OS: Pop!_OS 22.04
Browser: Firefox 123.0.1

### Grafana platform?

Docker

**Comment by tonypowa** [comment]
hi @retrodaredevil 

Are there any error messages or logs in the browser's console when the navigation occurs after clicking the link?

**Comment by Clarity-89** [comment]
It looks like the href's for URLs are sanitized in Grafana and that's why it doesn't behave the way you'd expect. You can use `href='#'` to prevent link navigation, however if you only need this for the link styles, I'd suggest creating a non-link component with the link's styles for accessibility reasons, among others.

**Comment by retrodaredevil** [comment]
Yeah I figured there was some sanitation going on somewhere, but I disagree that this should have been something for me to work around in the first place. This is known behavior in all browsers that those types of links don't navigate away from the page.

It might not be worth adding a fix for if this is the first time it has come up, because I don't think using `href="javascript:void 0"` is the best solution, but this was annoying to work around when there was code written using `href="javascript:void 0` and I had to make a PR to fix it just so it would work in Grafana.
